psc-ide-server/Main.hs

```diff
@@ -101,32 +103,43 @@ startServer port env = withSocketsDo $ do
 
     loop :: (PscIde m, MonadLogger m) => Socket -> m ()
     loop sock = do
-      (cmd,h) <- acceptCommand sock
-      case decodeT cmd of
-        Just cmd' -> do
-          result <- runExceptT (handleCommand cmd')
-          $(logDebug) ("Answer was: " <> T.pack (show result))
-          liftIO (hFlush stdout)
-          case result of
-            -- What function can I use to clean this up?
-            Right r  -> liftIO $ T.hPutStrLn h (encodeT r)
-            Left err -> liftIO $ T.hPutStrLn h (encodeT err)
-        Nothing -> do
-          $(logDebug) ("Parsing the command failed. Command: " <> cmd)
-          liftIO $ do
-            T.hPutStrLn h (encodeT (GeneralError "Error parsing Command."))
-            hFlush stdout
-      liftIO (hClose h)
+      accepted <- runExceptT $ acceptCommand sock
+      case accepted of
+        Left err -> $(logDebug) err
```

Comment:
In this case, if the debug flag is not set, won't this error just be ignored? Is that a problem?